### PR TITLE
Feat/restake unstake

### PR DESCRIPTION
**Pull request by rcstanciu**
- Reset APY before restaking a position
- Fetch the latest blockchain data on restake
- Created an API for waiting for a certain block to be indexed before returning: `GET /wait-for-block?block=61626364`

### CODE REVIEW

File: views/chain.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    Will be called as `GET /wait-for-block?block=123`,
+    where 123 is the block we need to wait for.
+    """
+    with Session() as s:
+        block = request.args.get("block")
+
+        if not block:
+            return {"ok": False, "error": "Block param is mandatory."}
+
+        block = int(block)
```
Comment (Evert0x) [inline views/chain.py:22]: what if parsing fails?

File: views/chain.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+        # Maximum number of seconds to wait for the block to get indexer.
+        # Nginx's timeout should be changed according to this setting as well.
+        max_tries = 60
+        try_count = 0
+
+        while try_count < max_tries:
```
Comment (Evert0x) [inline views/chain.py:29]: try_count is not incrementing?

File: views/chain.py (hunk 1 above)
Comment (rcstanciu) [reply views/chain.py:22]: It will throw an error that will get translated into an HTTP 500 error.

Comment (Evert0x) [reply views/chain.py:22]: Could be nice to throw a JSON error as well, up yo you.

File: views/chain.py (hunk 2 above)
Comment (rcstanciu) [reply views/chain.py:29]: Thanks for catching this!

File: views/chain.py (hunk 1 above)
Comment (rcstanciu) [reply views/chain.py]: Added JSON error response message and also different error codes (for when we will add frontend retries based on error codes).

**Review by Evert0x — APPROVED** [review]

**Pull request merged**